packages/components/src/mobile/bottom-sheet/cell.native.js

```diff
@@ -240,24 +265,27 @@ class BottomSheetCell extends Component {
 			// We also show the TextInput to display placeholder.
 			const shouldShowPlaceholder = isValueEditable && value === '';
 			return this.state.isEditingValue || shouldShowPlaceholder ? (
-				<TextInput
-					ref={ ( c ) => ( this._valueTextInput = c ) }
-					numberOfLines={ 1 }
-					style={ finalStyle }
-					value={ value }
-					placeholder={ valuePlaceholder }
-					placeholderTextColor={ '#87a6bc' }
-					onChangeText={ onChangeValue }
-					editable={ isValueEditable }
-					pointerEvents={
-						this.state.isEditingValue ? 'auto' : 'none'
-					}
-					onFocus={ startEditing }
-					onBlur={ finishEditing }
-					onSubmitEditing={ onSubmit }
-					keyboardType={ this.typeToKeyboardType( type, step ) }
-					{ ...valueProps }
-				/>
+				<View style={ styles.cellTextInputContainer }>
+					<TextInput
+						ref={ ( c ) => ( this._valueTextInput = c ) }
+						numberOfLines={ 1 }
+						style={ finalStyle }
+						value={ this.state.localValue }
+						placeholder={ valuePlaceholder }
+						placeholderTextColor={ '#87a6bc' }
+						onChangeText={ ( text ) => setTextInputValue( text ) }
+						editable={ isValueEditable }
+						pointerEvents={
+							this.state.isEditingValue ? 'auto' : 'none'
+						}
+						onFocus={ startEditing }
+						onBlur={ finishEditing }
+						onSubmitEditing={ onSubmit }
+						keyboardType={ this.typeToKeyboardType( type, step ) }
+						{ ...valueProps }
+					/>
+					{ displayClearButton && clearButton() }
```

Comment:
The `clearButton` function checks whether the local state `localValue` is empty to display the button, I'm wondering if we should include that condition here instead of within the function, wdyt?

https://github.com/WordPress/gutenberg/blob/0f6c1941d06d0095409ba8294089ef79427e6ea8/packages/components/src/mobile/bottom-sheet/cell.native.js#L205

Reply:
Good call, updated!